### ISSUE TITLE
Add new WarrantyType Field to assets model

### ISSUE THREAD

**Issue by uck9** [ArnesSI/netbox-inventory#134]
Wondering what reception would be had to suggesting the addition of a text field (or a link to a separate WarrantyType Model) for Warranty Type alongside the Warranty Start and End dates.  Thinking here around Cisco who have a number of different Warranty Types for Hardware separate to any support contracts.

**Comment by matejv** [comment]
If we implement Contracts as we're discussing in #130 then we could assume Asset warranty_start & warranty_end represents hardware warranty. Any support contracts, software contracts & such would be handled by assigning multiple contracts to asset. Agree?

My thinking is I'd leave asset warranty start / end as they are for users that don't need anything more than that. For those that need more granular support / warranty info for their assets, they could create contracts and assign them to assets as needed.

**Comment by uck9** [comment]
Agreed on keeping this seperate to any discussions around support contracts.  Suggesting to add a hardware warranty type to the current model to better track the actual type of hardware warranty.  

As an example, Cisco have a multitude of types (https://www.cisco.com/c/en/us/products/warranty-listing.html#~warranty-finder-and-faq) depending on the model of hardware.  We’d be pulling this warranty type data in from the Cisco support api’s along with start and end dates and would be great to be able to include in the one screen in netbox.

Having this as a seperate model for warranty types (would mean you could include any doc url’s etc), but would be just as happy at this point to just have it as a free text field.

Does this help explain my thoughts better?
